### PR TITLE
[ToB][Frontend] Gradle tasks for ToB frontend

### LINKED ISSUE #24563: [Task]: Gradle tasks for ToB frontend
### What needs to happen?

The tasks should mirror the tasks of Playground frontend. In particular:

- `generate`
- `generateCode`
- `pubGet`
- `format`
- `run`
- `test`
- `precommit`
- `cleanFlutter`
- `cleanGenerated`

Some of the Playground's tasks calls appropriate tasks on `playground_components`, and this is easy because the latter is a sub-project of the former.

ToB also depends on `playground_components`, so ToB's tasks should call `playground_components`'s tasks in all the same cases that Playground does. This may be a bit harder because that is not a sub-project anymore.

The Gradle project file for ToB should be in `.kts` format.

### Issue Priority

Priority: 3

### Issue Component

Component: beam-playground

### PR DESCRIPTION
**Pull request by nausharipov**
Resolves #24563 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by nausharipov** [comment]
R: @alexeyinkin

File: learning/tour-of-beam/frontend/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+tasks.register("analyze") {
+  dependsOn(":playground:frontend:playground_components:generateCode")
+  dependsOn("generateCode")
+
+  group = "verification"
+  description = "Analyze dart code"
+
+  doLast {
+    exec {
+      // Exact paths instead of '.' so it does not go into playground_components
```
Comment (alexeyinkin) [inline learning/tour-of-beam/frontend/build.gradle.kts:39]: Delete such comments here and in Playground so we can copy tasks between them.

File: learning/tour-of-beam/frontend/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+      executable("dart")
+      args("analyze", "lib", "test")
+    }
+  }
+}
+
+tasks.register("pubGet") {
+  group = "build"
+  description = "Get packages for the playground frontend project"
```
Comment (alexeyinkin) [inline learning/tour-of-beam/frontend/build.gradle.kts:48]: Strip project names from descriptions so the same ones can be used for both Playground and ToB.

**Review by alexeyinkin — CHANGES_REQUESTED** [review]

**Comment by nausharipov** [comment]
A test fails in `playground_controller_test.dart`. Does it need to be fixed in this PR? @alexeyinkin 

Screenshot 2023-01-06 at 15 16 29

**Review by alexeyinkin — APPROVED** [review]
LGTM (internal review).

**Comment by nausharipov** [comment]
R: @damondouglas

**Comment by nausharipov** [comment]
The README will be updated in a separate task: https://github.com/apache/beam/issues/25076

**Review by damondouglas — APPROVED** [review]
@damccorm LGTM

**Review by damccorm — APPROVED** [review]

**Pull request merged**